modules/core/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/messages/GridQueryNextPageResponse.java

```diff
@@ -39,42 +34,50 @@ public class GridQueryNextPageResponse implements Message, Serializable {
     private static final long serialVersionUID = 0L;
 
     /** */
+    @Order(value = 0, method = "queryRequestId")
     private long qryReqId;
 
     /** */
+    @Order(1)
     private int segmentId;
 
     /** */
+    @Order(value = 2, method = "query")
     private int qry;
 
     /** */
+    @Order(3)
     private int page;
 
     /** */
+    @Order(4)
     private int allRows;
 
     /** */
+    @Order(value = 5, method = "columns")
     private int cols;
 
     /** */
-    @GridDirectCollection(Message.class)
+    @Order(value = 6, method = "values")
     private Collection<Message> vals;
 
     /**
      * Note, columns count in plain row can differ from {@link #cols}.
      * See {@code org.apache.ignite.internal.processors.query.h2.twostep.msg.GridH2ValueMessageFactory#toMessages}.
      * See javadoc for {@code org.h2.result.ResultInterface#getVisibleColumnCount()} and {@code org.h2.result.ResultInterface#currentRow()}.
      */
-    @GridDirectTransient
     private transient Collection<?> plainRows;
```

Comment:
Do we need transient here? Which reason for Serializable and `IgniteCodeGeneratingFail` annotation? Let's refactor this class in a separate patch